tests/properties/pflua_pipelines_match.lua

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env luajit
+-- -*- lua -*-
+module(..., package.seeall)
+package.path = package.path .. ";../?.lua;../../src/?.lua"
+-- Compare the results of the libpcap/bpf and pure-lua pflua pipelines.
+
+local pf = require("pf")
+local savefile = require("pf.savefile")
+local utils = require('pf.utils')
+
+local pflang = require('pfquickcheck.pflang')
+
+function property(packets)
+   --nil pkt_idx, pflang_expr, bpf_result, pflua_result to avoid
+   -- confusing debug information
+   pkt_idx, pflang_expr, bpf_result, pflua_result = nil
+   local pkt, P, pkt_len, libpcap_pred, pflua_pred
+   a = pflang.Pflang()
+   pflang_expr = table.concat(a, ' ')
```

Comment:
I don't understand.  If you are intending to generate only valid pflang, you need to handle grouping.  The best way, the way that will most closely reflect what users write, is to reason about precedence to know when parens are needed and when they aren't.  An easy first way would be to just add parens everywhere.